capsules/extra/src/ieee802154/driver.rs

```diff
@@ -505,15 +518,22 @@ impl framer::DeviceProcedure for RadioDriver<'_> {
     /// Gets the long address corresponding to the neighbor that matches the given
     /// MAC address. If no such neighbor exists, returns `None`.
     fn lookup_addr_long(&self, addr: MacAddress) -> Option<[u8; 8]> {
-        self.neighbors.and_then(|neighbors| {
+        let res = self.neighbors.and_then(|neighbors| {
             neighbors[..self.num_neighbors.get()]
                 .iter()
                 .find(|neighbor| match addr {
                     MacAddress::Short(addr) => addr == neighbor.short_addr,
                     MacAddress::Long(addr) => addr == neighbor.long_addr,
                 })
                 .map(|neighbor| neighbor.long_addr)
-        })
+        });
+        if res.is_none() {
+            self.backup_device_procedure
+                .unwrap_or_panic()
```

Comment:
Can you give some more details here, maybe in a comment. Why is this panic okay? What does it mean to do a "backup procedure"? Is this something that is quite possible to fail at runtime, or is this supposed to always succeed, unless there's an implementation issue?

Reply:
The device backup procedure arose due to the current design of 15.4. The 15.4 driver stores the network keys and the framer looks them up accordingly using a lookup procedure trait (when receiving/sending a 15.4 packet). Once joined to the Thread network, Thread messages are encrypted within the UDP payload and also at the link layer. Because of this, we need to possess a way to add to the Radio Driver's stored keys. My solution was to add a "backup lookup" so that if a key was not found, the 15.4 framer would check for the Thread key next. Long term it is not ideal for the 15.4 driver to hold these keys (since it makes it rather clunky for other capsules to add/remove keys).

This is something I will keep thinking about. In the short term, my fix should be adequet and does not cause much disruption to the 15.4 capsule. I will go ahead and remove the `panic` because it should likely just return an error if this is not set during the configuration. 

I will add some comments and TODOs for this so we have a note moving forward.